cmd/kubeadm/app/master/selfhosted_volumes.go

```diff
@@ -0,0 +1,332 @@
+									Path: path.Join(path.Base(cfg.CertificatesDir), kubeadmconstants.FrontProxyCACertName),
+								},
+							},
+						},
+					},
+					{
+						Secret: &v1.SecretProjection{
+							LocalObjectReference: v1.LocalObjectReference{
+								Name: kubeadmconstants.FrontProxyClientCertAndKeyBaseName,
+							},
+							Items: []v1.KeyToPath{
+								{
+									Key:  v1.TLSCertKey,
+									Path: path.Join(path.Base(cfg.CertificatesDir), kubeadmconstants.FrontProxyClientCertName),
+								},
+								{
+									Key:  v1.TLSPrivateKeyKey,
+									Path: path.Join(path.Base(cfg.CertificatesDir), kubeadmconstants.FrontProxyClientKeyName),
+								},
+							},
+						},
+					},
+				},
+			},
+		},
+	}
+}
+
+func schedulerProjectedVolume() v1.Volume {
+	return v1.Volume{
+		Name: "k8s",
+		VolumeSource: v1.VolumeSource{
+			Projected: &v1.ProjectedVolumeSource{
+				Sources: []v1.VolumeProjection{
+					{
+						Secret: &v1.SecretProjection{
+							LocalObjectReference: v1.LocalObjectReference{
+								Name: kubeadmconstants.SchedulerKubeConfigFileName,
+							},
+						},
+					},
+				},
+			},
+		},
+	}
+}
+
+func controllerManagerProjectedVolume(cfg *kubeadmapi.MasterConfiguration) v1.Volume {
+	return v1.Volume{
+		Name: "k8s",
+		VolumeSource: v1.VolumeSource{
+			Projected: &v1.ProjectedVolumeSource{
+				Sources: []v1.VolumeProjection{
+					{
+						Secret: &v1.SecretProjection{
+							LocalObjectReference: v1.LocalObjectReference{
+								Name: kubeadmconstants.ControllerManagerKubeConfigFileName,
+							},
+						},
+					},
+					{
+						Secret: &v1.SecretProjection{
+							LocalObjectReference: v1.LocalObjectReference{
+								Name: kubeadmconstants.CACertAndKeyBaseName,
+							},
+							Items: []v1.KeyToPath{
+								{
+									Key:  v1.TLSCertKey,
+									Path: path.Join(path.Base(cfg.CertificatesDir), kubeadmconstants.CACertName),
+								},
+								{
+									Key:  v1.TLSPrivateKeyKey,
+									Path: path.Join(path.Base(cfg.CertificatesDir), kubeadmconstants.CAKeyName),
+								},
+							},
+						},
+					},
+					{
+						Secret: &v1.SecretProjection{
+							LocalObjectReference: v1.LocalObjectReference{
+								Name: kubeadmconstants.ServiceAccountKeyBaseName,
+							},
+							Items: []v1.KeyToPath{
+								{
+									Key:  v1.TLSPrivateKeyKey,
+									Path: path.Join(path.Base(cfg.CertificatesDir), kubeadmconstants.ServiceAccountPrivateKeyName),
+								},
+							},
+						},
+					},
+				},
+			},
+		},
+	}
+}
+
+func createTLSSecrets(cfg *kubeadmapi.MasterConfiguration, client *clientset.Clientset) error {
+	for _, tlsKeyPair := range getTLSKeyPairs() {
+		secret, err := createTLSSecretFromFiles(
+			tlsKeyPair.name,
+			path.Join(cfg.CertificatesDir, tlsKeyPair.cert),
+			path.Join(cfg.CertificatesDir, tlsKeyPair.key),
+		)
+		if err != nil {
+			return err
+		}
+
+		if _, err := client.CoreV1().Secrets(metav1.NamespaceSystem).Create(secret); err != nil {
```

Comment:
So I looked through the default policies and TBH I don't think we'd have a good answer any time soon, I think a prudent measure would be to add a variable whose default is currently the  systemnamespace to allow for future over-rides.